llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -5045,6 +5045,44 @@ void cs6475_debug(std::string DbgString) {
     dbgs() << DbgString;
 }
 
+Instruction *cs6475_optimizer_suraj(Instruction *I) {
+  // Converts
+  // if x is even, return x+1 else return x-1
+  // to
+  // return x^1
+
+  Value *INPUT, *A, *B, *C;
+
+  if (!match(I, m_c_Add(m_Value(C), m_Value(INPUT)))) {
+    return nullptr;
+  }
+  cs6475_debug("Add Expression Matched\n");
+
+  if (!match(C, m_Select(m_Value(B), m_One(), m_ConstantInt<-1>()))) {
```

Comment:
LLVM integers are arbitrary width but your -1 is a 32-bit constant, I'm pretty sure this will fail for large LLVM integers, please create a -1 in a portable fashion